e2e/e2e_test.go

```diff
@@ -450,36 +485,39 @@ func Test_EVM_Hedera_Native_Token(t *testing.T) {
 		setupEnv.Clients.Hedera.GetOperatorAccountID().String(),
 		status.Completed,
 		evm.Signer.Address(),
-		entity.NanoTime{Time: blockTimestamp})
+		entity.NanoTime{Time: blockTimestamp},
+	)
 
-	expectedScheduleMintRecord := &entity.Schedule{
-		TransactionID: bridgeMintTransactionID,
-		ScheduleID:    bridgeMintScheduleID,
-		Operation:     schedule.MINT,
-		Status:        status.Completed,
-		TransferID: sql.NullString{
+	expectedScheduleMintRecord := expected.ScheduleRecord(
+		bridgeMintTransactionID,
+		bridgeMintScheduleID,
+		schedule.MINT,
+		false,
+		status.Completed,
+		sql.NullString{
 			String: lockEventId,
 			Valid:  true,
 		},
-	}
+	)
 
 	// Wait for validators to update DB state after Scheduled TX is mined
-	time.Sleep(30 * time.Second)
 
 	// Step 6: Verify that records have been created successfully
-	verifyTransferRecord(setupEnv.DbValidator, expectedLockEventRecord, t)
-	verifyScheduleRecord(setupEnv.DbValidator, expectedScheduleMintRecord, t)
+	verify.TransferRecord(t, setupEnv.DbValidator, expectedLockEventRecord)
+	verify.ScheduleRecord(t, setupEnv.DbValidator, expectedScheduleMintRecord)
 
 	// Step 7: Validate that a scheduled transfer txn was submitted successfully
-	bridgeTransferTransactionID, bridgeTransferScheduleID := validateScheduledTx(
-		setupEnv,
+	bridgeTransferTransactionID, bridgeTransferScheduleID := verify.ScheduledTx(
+		t,
+		setupEnv.Clients.Hedera,
+		setupEnv.Clients.MirrorNode,
 		setupEnv.Clients.Hedera.GetOperatorAccountID(),
 		setupEnv.TokenID.String(),
-		generateMirrorNodeExpectedTransfersForLockEvent(setupEnv, targetAsset, expectedAmount),
-		t)
+		expected.MirrorNodeExpectedTransfersForLockEvent(setupEnv.Clients.Hedera, setupEnv.BridgeAccount, targetAsset, expectedAmount),
+		now,
+	)
 
 	// Wait for validators to update DB state after Scheduled TX is mined
-	time.Sleep(30 * time.Second)
```

Comment:
I think we will need a more stable fix, as we already sync we could have a re-try in case of timeout.